bitbots_vision/scripts/evaluator.py

```diff
@@ -0,0 +1,738 @@
+        # It also skips if a lock is present
+        if (self._recieved_all_messages_for_image(self._current_image_counter) or timeout) and not self._lock:
+            # Increse image counter
+            self._current_image_counter += 1
+            # Reset timeout timer
+            self._image_send_time = rospy.Time.now()
+            # Send image
+            self._send_image()
+
+    def _get_image_name(self):
+        """
+        Returns the name of the currently processed image
+        """
+        return self._images[self._current_image_counter]['name']
+
+    def _get_current_labels(self):
+        """
+        Returns the annotations of the currently processed image
+        """
+        return self._images[self._current_image_counter]['annotations']
+
+    def _send_image(self):
+        """
+        Sends an Image to the Vision
+        """
+        # handle stop at end of image list
+        if self._current_image_counter >= self._image_count:  # iterated through all images
+            rospy.loginfo('iterated through all images.')
+            self._stop = True
+            return
+
+        # Get image name/id
+        name = self._get_image_name()
+
+        rospy.loginfo(f"Process image '{name}''")
+
+        # Read image file
+        imgpath = os.path.join(self._image_path, name)
+        image = cv2.imread(imgpath)
+        if image is None:
+            rospy.logwarn('Could not open image {} at path {}'.format(name, self._image_path))
+            return
+
+
+        # Setting label size as used by the imagetagger before we do the resizeing
+        if self._label_shape is None:
+            self._label_shape = image.shape[:-1]
+
+
+        # Set the resized image size as expected by the vision
+        if self._image_shape is None:
+            self._image_shape = np.array((
+                rospy.get_param("~image_resolution_y"), 
+                rospy.get_param("~image_resolution_x")))
+
+        # resize the image
+        image = cv2.resize(image, tuple(np.flip(self._image_shape)), interpolation=3)
+
+        # Building and sending message
+        rospy.loginfo('Sending image {} of {} (starting by 0).'.format(self._current_image_counter, self._image_count))
+        msg = self.bridge.cv2_to_imgmsg(image, 'bgr8')
+        msg.header.stamp = rospy.Time.now()
+        # The second unforgivable curse. 
+        # The tf frame name is abused to store the id of the image
+        msg.header.frame_id = str(self._current_image_counter)
+        self._image_pub.publish(msg)
+
+        # Set up evaluation element in measurements list
+        self._measurements[self._current_image_counter] = ImageMeasurement(self._images[self._current_image_counter], self._evaluated_classes)
+
+    def _read_labels(self, filename):
+        """
+        Reads the labels YAML file and returns a list of image names with their labels.
+        This is set up to work with the "evaluation" export format of the Bit-Bots Team in the ImageTagger.
```

Comment:
I suggest providing an demo-file to understand the structure (especially for non imagetagger users)